go-controller/pkg/ovn/ovn.go

```diff
@@ -151,3 +181,63 @@ func (oc *Controller) WatchEndpoints() {
 		},
 	})
 }
+
+func (oc *Controller) WatchNetworkPolicy() {
+	oc.StartPolicyWatch(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			policy := obj.(*kapisnetworking.NetworkPolicy)
+			oc.addNetworkPolicy(policy)
+			return
+		},
+		UpdateFunc: func(old, new interface{}) {
+			return
+		},
+		DeleteFunc: func(obj interface{}) {
+			policy, ok := obj.(*kapisnetworking.NetworkPolicy)
+			if !ok {
+				tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+				if !ok {
+					logrus.Errorf("couldn't get object from tombstone %+v", obj)
+					return
+				}
+				policy, ok = tombstone.Obj.(*kapisnetworking.NetworkPolicy)
+				if !ok {
+					logrus.Errorf("tombstone contained object that is not a pod %#v", obj)
+					return
+				}
+			}
+			oc.deleteNetworkPolicy(policy)
+			return
+		},
+	})
+}
+
+func (oc *Controller) WatchNamespaces() {
+	oc.StartNamespaceWatch(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			ns := obj.(*kapi.Namespace)
+			oc.addNamespace(ns)
+			return
+		},
+		UpdateFunc: func(old, new interface{}) {
```

Comment:
We should handle the UpdateNamespace case here. e.g. what if 'network-policy' annotation is added after creation of the namespace.